### PR TITLE
Add resource definitions to init containers

### PR DESCRIPTION
**Pull request by stranljip**
The init containers were lacking the possibility to add resource definitions. This PR adds the possibility to add them optionally.

### CODE REVIEW

File: charts/artifact-hub/templates/hub_deployment.yaml [hunk 1]
```diff
@@ -64,8 +68,10 @@ spec:
             - name: http
               containerPort: 8000
               protocol: TCP
+          {{- with .Values.hub.deploy.resources }}
```
Comment (tegioz) [inline charts/artifact-hub/templates/hub_deployment.yaml:71]: At the moment the `resources` value for the hub is [marked as required in the schema](https://github.com/artifacthub/hub/blob/c73dafa519020415927665e14fb6eac1066120eb/charts/artifact-hub/values.schema.json#L312). With this change, it won't be anymore. I'm fine switching to `with` for resources, but if we go this way we should probably change the rest of the resources entries in other templates (tracker, scanner, etc) and not mark them as required anymore. Does that make sense?

**Review by tegioz — CHANGES_REQUESTED** [review]
Hi @stranljip 👋 

Thanks for the PR!

File: charts/artifact-hub/templates/hub_deployment.yaml (hunk 1 above)
Comment (stranljip) [reply charts/artifact-hub/templates/hub_deployment.yaml:71]: This is probably a general decision - in our clusters we require the declaration of resources but I know that there are many cases where you explicitly do not want to have resource definitions. Personally I am in favour of letting the user decide what she/he wants, so I would argue to make all occurrences optional.
But - this is my first time contribution to this project so I feel that this should be your decision :-)

Comment (tegioz) [reply charts/artifact-hub/templates/hub_deployment.yaml:71]: Let's make them all optional then 👍

File: charts/artifact-hub/templates/_helpers.tpl [hunk 2]
```diff
@@ -93,6 +93,10 @@ Provide an init container to verify the database is accessible
 name: check-db-ready
 image: {{ .Values.postgresql.image.repository }}:{{ .Values.postgresql.image.tag }}
 imagePullPolicy: {{ .Values.pullPolicy }}
+{{- with .Values.hub.deploy.initContainers.checkDbIsReady.resources }}
```
[the remark this answers is not in the document]
Comment (stranljip) [reply charts/artifact-hub/templates/_helpers.tpl:96]: Done

File: charts/artifact-hub/templates/hub_deployment.yaml (hunk 1 above)
Comment (stranljip) [reply charts/artifact-hub/templates/hub_deployment.yaml:71]: Done by my last commit :-)

File: charts/artifact-hub/values.yaml [hunk 3]
```diff
@@ -119,12 +119,39 @@ hub:
     image:
       # Hub image repository (without the tag)
       repository: artifacthub/hub
+    # If you do want to specify resources, uncomment the following
```
Comment (tegioz) [inline charts/artifact-hub/values.yaml:122]: Thanks for adding those comments! Do you think it'd be clearer for users if we add them *below* the resources field? Just uncomment, remove braces and ready 😉

File: charts/artifact-hub/values.schema.json [hunk 4]
```diff
@@ -270,6 +270,24 @@
                                                     "default": ""
                                                 }
                                             }
+                                        },
+                                        "resources": {
+                                            "title": "Hub pod resource requirements",
```
Comment (tegioz) [inline charts/artifact-hub/values.schema.json:275]: Can you replace `Hub` for `DB migrator` or something similar? Hub is actually one of the components :) Same for the DB readiness one please.

**Review by tegioz — CHANGES_REQUESTED** [review]
Thanks for addressing those changes! I think this is almost ready to go. Can you please make sure the commits are signed so that the DCO check passes and rebase from master?

File: charts/artifact-hub/values.schema.json (hunk 4 above)
Comment (stranljip) [reply charts/artifact-hub/values.schema.json:275]: Argh ... sure, sorry.

Comment (tegioz) [reply charts/artifact-hub/values.schema.json:275]: Thanks, no problem 👍

File: charts/artifact-hub/values.yaml (hunk 3 above)
Comment (stranljip) [reply charts/artifact-hub/values.yaml:122]: I refactored/moved the comments.

File: charts/artifact-hub/values.schema.json (hunk 4 above)
Comment (stranljip) [reply charts/artifact-hub/values.schema.json]: I changed the titles to a context specific text

**Review by tegioz — APPROVED** [review]
Thanks @stranljip!

**Comment by tegioz** [comment]
I overlooked the issue the linter caught (https://github.com/artifacthub/hub/actions/runs/3083065453/jobs/4983549123) during my review, sorry.

I think we'll need some braces for `checkDbIsReady` in the values.yaml file @stranljip.

**Comment by stranljip** [comment]
> I think we'll need some braces for `checkDbIsReady` in the values.yaml file @stranljip.

I hope it is fine now - I just found out while testing without explicit values (sorry that I didn't do that before ...). I explicitly test for the occurrence of the element now, so the linter should be green now.

**Comment by tegioz** [comment]
Thanks, no worries 👍

**Comment by tegioz** [comment]
Merged! 🚀

**Pull request merged**